coderd/telemetry/telemetry.go

```diff
@@ -1753,6 +1767,54 @@ type Organization struct {
 	CreatedAt time.Time `json:"created_at"`
 }
 
+type Task struct {
+	ID                     string    `json:"id"`
+	OrganizationID         string    `json:"organization_id"`
+	OwnerID                string    `json:"owner_id"`
+	Name                   string    `json:"name"`
+	WorkspaceID            *string   `json:"workspace_id"`
+	WorkspaceBuildNumber   *int64    `json:"workspace_build_number"`
+	WorkspaceAgentID       *string   `json:"workspace_agent_id"`
+	WorkspaceAppID         *string   `json:"workspace_app_id"`
+	TemplateVersionID      string    `json:"template_version_id"`
+	TemplateParametersHash string    `json:"template_parameters_hash"` // Parameters are hashed for privacy.
+	PromptHash             string    `json:"prompt_hash"`              // Prompt is hashed for privacy.
+	CreatedAt              time.Time `json:"created_at"`
+	Status                 string    `json:"status"`
+}
+
+// ConvertTask anonymizes a Task.
+func ConvertTask(task database.Task) Task {
+	t := &Task{
+		ID:                     task.ID.String(),
+		OrganizationID:         task.OrganizationID.String(),
+		OwnerID:                task.OwnerID.String(),
+		Name:                   task.Name,
+		WorkspaceID:            nil,
+		WorkspaceBuildNumber:   nil,
+		WorkspaceAgentID:       nil,
+		WorkspaceAppID:         nil,
+		TemplateVersionID:      task.TemplateVersionID.String(),
+		TemplateParametersHash: fmt.Sprintf("%x", sha256.Sum256(task.TemplateParameters)),
```

Comment:
`{"foo": "1", "bar": "2"}` and `{"bar": "2", "foo": "1"}` are equivalent JSON values, but unless Postgres normalizes JSON when storing it, they would produce different hashes. Is that ok, or is it a bug?

Reply:
That's a good point. I don't think we have any ordering guarantees in our JSON serialization in `coderd/database`. It's fairly certain that PostgreSQL has some internal mechanism for computing a hash of a JSON/JSONB object, but I don't think it's exposed as a function or operator. We may just need to do this in code. Alternatively, we just drop this from the telemetry struct. I kind of added this on "auto-pilot" and now that I look more closely, it doesn't appear that we store any workspace parameters in telemetry snapshots. 

WDYT?

Reply:
https://github.com/coder/coder/pull/20279/commits/40b4cd6b65081723980a434579aaaa651e250d6d